File: assets/backing-services/typescript/tests/event-store-memory.test.ts
import { createInMemoryEventStore } from '../../src/adapters/driven/event-store-memory.js';
import { defaultTagsOf } from '../../src/application/ports/events.js';
import { eventStoreContract } from './event-store-contract.js';

/**
 * The contract, against the fake. This runs in `make verify` and needs no Docker, which is the whole
 * reason the fake exists. `make test-integration` runs the same contract against Postgres.
 */
eventStoreContract('the in-memory event store', async (tagsOf = defaultTagsOf) =>
  createInMemoryEventStore(tagsOf),
);

File: assets/backing-services/java/tests/event_store_memory_test.java
package com.example.deliverystarter.adapters.driven.eventstorememory;

import com.example.deliverystarter.application.ports.events.EventStore;
import com.example.deliverystarter.application.ports.events.TagsOf;
import com.example.deliverystarter.eventstorecontract.EventStoreContract;

/**
 * The contract, against the fake.
 *
 * <p>This runs in {@code make verify} and needs no Docker, which is the whole reason the fake exists.
 * {@code make test-integration} runs the same contract against Postgres.
 */
class InMemoryEventStoreTest extends EventStoreContract {

    @Override
    protected EventStore newStore(TagsOf tagsOf) {
        return new InMemoryEventStore(tagsOf, new InMemoryDatabase());
    }
}
